### ISSUE TITLE
Queries are kept waiting if result is too huge.

### ISSUE THREAD

**Issue by hey-jude** [getredash/redash#1829]
### Issue Summary

if some query has huge result, query browser is kept waiting.
But system status has normally done query.

### Technical details:

This is from sentry error report.
```
MemoryError: null
  File "flask/app.py", line 1639, in full_dispatch_request
    rv = self.dispatch_request()
  File "flask/app.py", line 1625, in dispatch_request
    return self.view_functions[rule.endpoint](**req.view_args)
  File "flask_restful/__init__.py", line 477, in wrapper
    resp = resource(*args, **kwargs)
  File "dist-packages/flask_login.py", line 758, in decorated_view
    return func(*args, **kwargs)
  File "flask/views.py", line 84, in view
    return self.dispatch_request(*args, **kwargs)
  File "redash/handlers/base.py", line 28, in dispatch_request
    return super(BaseResource, self).dispatch_request(*args, **kwargs)
  File "flask_restful/__init__.py", line 587, in dispatch_request
    resp = meth(*args, **kwargs)
  File "redash/permissions.py", line 47, in decorated
    return fn(*args, **kwargs)
  File "redash/handlers/query_results.py", line 165, in get
    response = self.make_json_response(query_result)
  File "redash/handlers/query_results.py", line 183, in make_json_response
    data = json.dumps({'query_result': query_result.to_dict()}, cls=utils.JSONEncoder)
  File "json/__init__.py", line 251, in dumps
    sort_keys=sort_keys, **kw).encode(obj)
  File "json/encoder.py", line 207, in encode
    chunks = self.iterencode(o, _one_shot=True)
  File "json/encoder.py", line 270, in iterencode
    return _iterencode(o, 0)
```

* Redash Version: 1.0.3
* Browser/OS: chrome / macos
* How did you install Redash:

**Comment by chauffer** [comment]
👍 getting this error as well, adding a `LIMIT 10` to the query makes it show; seems like a big issue.

Using postgres here.

**Comment by arikfr** [comment]
In previous versions the UI would keep showing the timer even if the request for a result failed. It was fixed in v2.0. Please reopen if you still experience this after upgrading.
